src/utils/zeroExUtils.ts

```diff
@@ -0,0 +1,143 @@
+import axios from 'axios'
+import querystring from 'querystring'
+import BigNumber from 'utils/bignumber'
+
+import { tokenInfo } from 'constants/tokenInfo'
+import { ZeroExData } from '../contexts/BuySell/types'
+import {
+  fetchCurrentPrice,
+  fetchHistoricalTokenMarketData,
+} from './coingeckoApi'
+
+export const getZeroExTradeData = async (
+  isExactInput: boolean,
+  isUserBuying: boolean,
+  currencyToken: string,
+  buySellToken: string,
+  buySellAmount: string
+): Promise<ZeroExData> => {
+  let sellToken
+  let buyToken
+
+  if (isUserBuying) {
+    buyToken = buySellToken
+    sellToken = currencyToken
+  } else {
+    buyToken = currencyToken
+    sellToken = buySellToken
+  }
+
+  const params = getApiParams(isExactInput, sellToken, buyToken, buySellAmount)
+  const resp = await axios.get(
+    `https://api.0x.org/swap/v1/quote?${querystring.stringify(params)}`
+  )
+
+  const zeroExData: ZeroExData = resp.data
+  return await processApiResult(
+    zeroExData,
+    isExactInput,
+    sellToken,
+    buySellToken,
+    buySellAmount
+  )
+}
+
+const getApiParams = (
+  isExactInput: boolean,
+  sellToken: string,
+  buyToken: string,
+  buySellAmount: string
+): any => {
+  const params: any = {
+    sellToken: tokenInfo[sellToken].address,
+    buyToken: tokenInfo[buyToken].address,
+    excludedSources: 'MultiHop', // MultiHop is usually not worth the gas price
+  }
+
+  if (isExactInput) {
+    params.sellAmount = getDecimalAdjustedAmount(
+      buySellAmount,
+      tokenInfo[sellToken].decimals
+    )
+  } else {
+    params.buyAmount = getDecimalAdjustedAmount(
+      buySellAmount,
+      tokenInfo[buyToken].decimals
+    )
+  }
+
+  return params
+}
+
+const processApiResult = async (
```

Comment:
can we add a comment in here detailing what this does? is it mostly formatting for people to read? does this affect the payload that we submit to 0x?